src/main/java/seedu/cafectrl/parser/Parser.java

```diff
@@ -192,23 +194,27 @@ private static Command prepareEditPriceCommand(Menu menu, String arguments, Ui u
             String dishIndexText = matcher.group(dishIndexGroup).trim();
 
             // Check whether the index is empty
-            if (dishIndexText.equals("")) {
+            if (dishIndexText.isEmpty()) {
+                logger.warning("Empty dish index!");
                 return new IncorrectCommand(ErrorMessages.MISSING_DISH_IN_EDIT_PRICE, ui);
             }
 
             dishIndex = Integer.parseInt(dishIndexText);
 
             // Check whether the dish index is valid
             if (!menu.isValidDishIndex(dishIndex)) {
+                logger.warning("Invalid dish index!");
                 return new IncorrectCommand(ErrorMessages.INVALID_DISH_INDEX, ui);
             }
         } catch (NumberFormatException e) {
+            logger.warning("Invalid dish index type!");
             return new IncorrectCommand(ErrorMessages.WRONG_DISH_INDEX_TYPE_FOR_EDIT_PRICE, ui);
         }
 
         try {
             newPrice = parsePriceToFloat(matcher.group(newPriceGroup).trim());
         } catch (ParserException e) {
+            logger.warning("Invalid price!");
```

Comment:
Perhaps you could pass in the exception thrown here too?
```suggestion
            logger.log(Level.WARNING, "Invalid price!", e);
```